File: lib/JFFUtils/JFFUtilsTest/NSObject+OwnershipsTest.m
#import "NSObject+OwnershipsTest.h"

@implementation NSObjectOwnershipsExtensionTest

- (void)testObjectOwnershipsExtension
{
    __weak NSObject *ownedDeallocated;
    {
        NSObject *owner = [NSObject new];

        {
            NSObject *owned = [NSObject new];
            
            ownedDeallocated = owned;
            
            [owner addOwnedObject:owned];
        }
        
        XCTAssertNotNil(ownedDeallocated, @"Owned should not be dealloced");
    }
    XCTAssertNil(ownedDeallocated, @"Owned should be dealloced");
}

@end

File: iAsync.Swift/lib/JUtils/JUtilsTests/NSObject+OwnershipsTest.swift
//
//  NSObject+OwnershipsTest.swift
//  JUtilsTests
//
//  Created by Vladimir Gorbenko on 14.06.14.
//  Copyright (c) 2014 EmbeddedSources. All rights reserved.
//

import XCTest

class NSObject_OwnershipsTest: XCTestCase {
    
    func testObjectOwnershipsExtension() {
        
        weak var ownedDeallocated: NSObject?
        
        { () -> () in
            let owner = NSObject()
            
            { () -> () in
                let owned = NSObject()
                ownedDeallocated = owned
                owner.addOwnedObject(owned)
            }()
            
            XCTAssertNotNil(ownedDeallocated, "Owned should not be dealloced")
        }()
        
        XCTAssertNil(ownedDeallocated, "Owned should be dealloced")
    }
}
